packages/workspace/src/generators/convert-to-monorepo/convert-to-monorepo.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  convertNxGenerator,
+  getProjects,
+  joinPathFragments,
+  ProjectConfiguration,
+  readNxJson,
+  Tree,
+  updateNxJson,
+} from '@nx/devkit';
+import { moveGenerator } from '../move/move';
+import { Schema } from './schema';
+
+export async function monorepoGenerator(tree: Tree, options: Schema) {
+  const projects = getProjects(tree);
+  mayebExtractTsConfigBase(tree);
+  await maybeExtractJestConfigBase(tree);
+
+  const nxJson = readNxJson(tree);
+  nxJson.workspaceLayout = {
+    // normalize paths without trailing slash
+    appsDir: joinPathFragments(options.appsDir),
+    libsDir: joinPathFragments(options.libsDir),
+  };
+
+  updateNxJson(tree, nxJson);
```

Comment:
Just add the directories and it will have the same effect. Do not write to nx.json.

This can move after the move.

1. If we move an app, we also have to add the libs dir that the user wants
2. If we move a lib, we don't actually have to do anything because moving the project will create the libs dir